Lib/multiprocessing/shared_memory.py

```diff
@@ -8,34 +8,120 @@
 
 from functools import reduce
 import mmap
-from .managers import DictProxy, SyncManager, Server
+from .managers import dispatch, BaseManager, Server, State, ProcessError, \
+                      BarrierProxy, AcquirerProxy, ConditionProxy, EventProxy
 from . import util
 import os
 import random
 import struct
 import sys
+import threading
 try:
-    from _posixshmem import _PosixSharedMemory, Error, ExistentialError, O_CREX
+    from _posixshmem import _PosixSharedMemory, \
+                            Error, ExistentialError, PermissionsError, \
+                            O_CREAT, O_EXCL, O_CREX, O_TRUNC
 except ImportError as ie:
     if os.name != "nt":
         # On Windows, posixshmem is not required to be available.
         raise ie
     else:
         _PosixSharedMemory = object
-        class ExistentialError(BaseException): pass
-        class Error(BaseException): pass
-        O_CREX = -1
+        class Error(Exception): pass
+        class ExistentialError(Error): pass
+        O_CREAT, O_EXCL, O_CREX, O_TRUNC = 64, 128, 192, 512
+
+if os.name == "nt":
+    import ctypes
+    from ctypes import wintypes
+
+    #kernel32 = ctypes.wintypes.WinDLL("kernel32", use_last_error=True)
```

Comment:
```suggestion
    # kernel32 = ctypes.wintypes.WinDLL("kernel32", use_last_error=True)
```

Reply:
Removed in da7731db412a94d9f06a99500b31c9147e601e3e.